src/main/java/com/amazon/ion/impl/bin/IonRawBinaryWriter_1_1.kt

```diff
@@ -468,37 +615,77 @@ class IonRawBinaryWriter_1_1 internal constructor(
 
     override fun stepInExpressionGroup(delimited: Boolean) {
         confirm(numAnnotations == 0) { "Cannot annotate an expression group" }
-        confirm(currentContainer.type == Macro) { "Can only create an expression group in a macro invocation" }
-        currentContainer = containerStack.push { it.reset(ExpressionGroup, buffer.position(), delimited) }
-        if (delimited) {
+        confirm(currentContainer.type == EEXP) { "Can only create an expression group in a macro invocation" }
+
+        val encoding = presenceBitmapStack.peek().signature[currentContainer.numChildren].type
+
+        currentContainer = containerStack.push { it.reset(EXPR_GROUP, buffer.position(), delimited, metadataOffset = 0) }
+        currentContainer.primitiveType = encoding.primitiveType
+
+        if (encoding.primitiveType != null) {
+            // Tagless groups always need a length (although it is actually the count of expressions in the group)
+            buffer.reserve(maxOf(1, lengthPrefixPreallocation))
```

Comment:
On the read side, for tagless groups, I interpreted the length as the number of bytes in the page, not the number of expressions. I think this simplifies handling of groups of variable-width tagless expressions.

Reply:
I thought we said number of expressions. I don't have a strong opinion as long as we're all doing the same thing. @zslayton, any thoughts?

Reply:
It's meant to be byte length per page.